src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -5914,33 +5914,32 @@ .locals init (bool V_0)
       IL_001c:  ldc.i4.0
       IL_001d:  stloc.0
       IL_001e:  ldloc.0
-      IL_001f:  brtrue.s   IL_0024
-      IL_0021:  ldc.i4.0
-      IL_0022:  br.s       IL_0025
-      IL_0024:  ldc.i4.1
-      IL_0025:  ret
+      IL_001f:  ret
     }
 ");
             compVerifier.VerifyIL("C.M2", @"
     {
-      // Code size       25 (0x19)
+      // Code size       30 (0x1e)
```

Comment:
Do we see the same size increase for these patterns? 

```csharp
static int M3(bool b1, bool b2) => (b1 || b2) ? 1 : 0;
static int M4(bool b1, bool b2) => (b1 && b2) ? 0 : 1;
```

It looks like the change is happening because the short circuit case the following pattern is needed (can be 1 or 0 depending on the condition):

``` 
ldc.i4.1
ret
```

Before your optimization that instruction sequence existed hence it could be re-used for both the short circuit and the non-short circuit case. Now that the `cgt` is present though there is no-reuse possible and more IL is needed.

Reply:
> Do we see the same size increase for these patterns?

No, for these I see improvements:

```diff
 var source = """
     class C
     {
         static int M1(bool b1, bool b2) => (b1 || b2) ? 1 : 0;
         static int M2(bool b1, bool b2) => (b1 && b2) ? 0 : 1;
     }
     """;
 var verifier = CompileAndVerify(source);
 verifier.VerifyDiagnostics();
 verifier.VerifyMethodBody("C.M1", """
     {
-      // Code size        9 (0x9)
+      // Code size        4 (0x4)
       .maxstack  2
       // sequence point: (b1 || b2) ? 1 : 0
       IL_0000:  ldarg.0
       IL_0001:  ldarg.1
       IL_0002:  or
-      IL_0003:  brtrue.s   IL_0007
-      IL_0005:  ldc.i4.0
-      IL_0006:  ret
-      IL_0007:  ldc.i4.1
-      IL_0008:  ret
+      IL_0003:  ret
     }
     """);
 verifier.VerifyMethodBody("C.M2", """
     {
-      // Code size        9 (0x9)
+      // Code size        7 (0x7)
       .maxstack  2
       // sequence point: (b1 && b2) ? 0 : 1
       IL_0000:  ldarg.0
       IL_0001:  ldarg.1
       IL_0002:  and
-      IL_0003:  brtrue.s   IL_0007
-      IL_0005:  ldc.i4.1
+      IL_0003:  ldc.i4.0
+      IL_0004:  ceq
       IL_0006:  ret
-      IL_0007:  ldc.i4.0
-      IL_0008:  ret
     }
     """);
```